### PR TITLE
some fixes in expressions

### PR DESCRIPTION
**Pull request by filiatra**

### CODE REVIEW

File: src/gsAssembler/gsExprAssembler.h [hunk 1]
```diff
@@ -779,6 +781,70 @@ void gsExprAssembler<T>::assemble(expr... args)
     m_matrix.makeCompressed();
 }
 
+template<class T>
+#if(__cplusplus >= 201103L) // c++11
+template<class... expr>
+void gsExprAssembler<T>::assemble(const bcRefList & BCs, expr... args)
+#else
+template <class E1, class E2, class E3, class E4, class E5>
+void gsExprAssembler<T>::assemble(const bcRefList & BCs, const expr::_expr<E1> & a1, const expr::_expr<E2> & a2, const expr::_expr<E3> & a3, const expr::_expr<E4> & a4, const expr::_expr<E5> & a5)
```
Comment (stefantakacs) [inline src/gsAssembler/gsExprAssembler.h]: What happens for less than 5 entries?

File: src/gsAssembler/gsExprAssembler.h [hunk 2]
```diff
@@ -344,8 +345,9 @@ class gsExprAssembler
     /// \sa gsExprAssembler::setIntegrationElements
     template<class... expr> void assemble(expr... args);
 
+    template<class... expr> void assemble(const bcRefList & BCs, expr... args);
+
     /*
-      template<class... expr> void assemble(const bcContainer & BCs, expr... args);
       template<class... expr> void assemble(const ppContainer & iFaces, expr... args);
     */
 #else
```
Comment (stefantakacs) [inline src/gsAssembler/gsExprAssembler.h]: line 358: why is this different to 355 o 362?

File: src/gsAssembler/gsExprAssembler.h (hunk 1 above)
Comment (filiatra) [reply src/gsAssembler/gsExprAssembler.h]: it works, (but not for more than 5)

File: src/gsAssembler/gsExprAssembler.h [hunk 3]
```diff
@@ -129,8 +129,10 @@ class gsExprAssembler
 
     typedef typename gsSparseMatrix<T>::BlockView matBlockView;
 
+    typedef typename gsBoundaryConditions<T>::bcRefList   bcRefList;
     typedef typename gsBoundaryConditions<T>::bcContainer bcContainer;
-    typedef typename gsBoundaryConditions<T>::ppContainer ppContainer;
+    //typedef typename gsBoundaryConditions<T>::ppContainer ppContainer;
+    typedef gsBoxTopology::ifContainer ppContainer;
```
Comment (stefantakacs) [inline src/gsAssembler/gsExprAssembler.h]: what pp stands for? couldn't it be ifContainer?

**Pull request merged**